mkosi

```diff
@@ -1871,6 +1871,10 @@ def install_grub(args, workspace, loopdev, grub):
             "--modules=ext2 part_gpt", "--target=i386-pc",
             loopdev, nspawn_params=nspawn_params)
 
+    # HACK: systemd-nspawn bug? Without this it fails with
+    # bind(/run/systemd/nspawn/notify) failed: Address already in use
+    unlink_try_hard(os.path.join(os.path.join(workspace, "root", "run/systemd/nspawn/notify")))
```

Comment:
Yep, I can confirm that I get the same issue without this HACK.

But I'd prefer that we merge this without this patch, and figure out what is wrong in `systemd-nspawn` and patch it there.

Reply:
yes, I can remove this just before merging.

Reply:
hu? /run in the container should be mounted from tmpfs anyway, so why would the dir in the workspace actually work? /me is very confused...

which nspawn version is that? something very recent?

Reply:
See more comments below... Sneaking a `ls` around the call to systemd-boot it's possible to see the file "leaks" outside. Version is the one in F29, but the commit hash in the package doesn't exist in systems repo. I hadn't this problem last week so it's probably between 239 and this one in Fedora.

Reply:
I messed up the github push. The version in F29/F30 is exactly the same as systemd-stable/v239-stable tip now. The right thing was to rebuild the package in Fedora, but it takes a while and I hoped nobody would notice ;)

Reply:
@keszybz now you probably have a reason to rebuild the package. I seems like the problem was a backport to stable.

You need either this: https://gist.github.com/cd092aa30f58e5ff2ccde6ef4b725fce

Or to bring https://github.com/systemd/systemd/commit/44ed5214add to stable. The problem was created when https://github.com/systemd/systemd/commit/c192917cab7 was backported without the other commit... so `sa.un.sun_path` would be NULL and we would end up not unlinking the socket.

Reply:
This is solved by backporting said commit to systemd-nspawn - tested on F29 with the additional commit and works fine.